Add "state x days ago" query (#27028)
home-assistant/home-assistant.github.io: source/_integrations/sql.markdown

```
Use `state` as column for value.

### State of an entity x time ago

> If you want to extract the state of entity from a day, hour or minute ago, the query is:

Reviewer: ```suggestion
If you want to extract the state of an entity from a day, hour, or minute ago, the query is:

```

```sql
SELECT state 
FROM states 
INNER JOIN states_meta ON states.metadata_id = states_meta.metadata_id
WHERE states_meta.entity_id = 'sensor.temperature_in' 
  AND last_updated_ts <= strftime('%s', 'now', '-1 day')

> ORDER BY last_updated_ts DESC LIMIT 1;

Reviewer: Can we format the SQL query just like all other queries on this page? It will help with readability.

Author: I lack knowledge about query structure to make correct indentations myself

Author: Done

```

> Replace  `-1 day` with the target offset, e.g. `-1 hour`.

Reviewer: ```suggestion

Replace `-1 day` with the target offset, for example, `-1 hour`.
```

```
Use `state` as column for value.

### State of an entity x time ago

If you want to extract the state of an entity from a day, hour, or minute ago, the query is:

```sql
SELECT 

>   state 

Reviewer 2: ```suggestion
  states.state
```

FROM 
  states 
  INNER JOIN states_meta ON 
    states.metadata_id = states_meta.metadata_id
WHERE 
  states_meta.entity_id = 'sensor.temperature_in' 
  AND last_updated_ts <= strftime('%s', 'now', '-1 day')
ORDER BY 
  last_updated_ts DESC 

> LIMIT 1;

Reviewer 2: ```suggestion
LIMIT
  1;
```
